extensions/amp-story/1.0/amp-story.js

```diff
@@ -410,8 +410,15 @@ export class AmpStory extends AMP.BaseElement {
     // prerendering, because of a height incorrectly set to 0.
     this.mutateElement(() => {});
 
-    const pageEl = this.element.querySelector('amp-story-page');
-    pageEl && pageEl.setAttribute('active', '');
+    const firstPage = this.element.querySelector(`amp-story-page`);
+    const activePage = firstPage
+      ? this.element.querySelector(
+          `amp-story-page#${escapeCssSelectorIdent(
+            this.getInitialPageId_(firstPage)
+          )}`
+        )
+      : null;
+    activePage && activePage.setAttribute('active', '');
```

Comment:
I think this all breaks the `layoutCallback` which won't resolve in a prerendering scenario.

Here, we assume that the first page is the only one that can be prerendered:

https://github.com/ampproject/amphtml/blob/19c934b4bc04c008121759b2cc61e6f7f44cb679/extensions/amp-story/1.0/amp-story.js#L1035-L1042

Instead, you should move the querySelector to the `getInitialPageId_()` method, and update the snippet I just linked to so it waits for the actual first page.

Reply:
I don't think it won't resolve because the first page also gets prerendered, but waiting for the active page could be a good idea. That's what you meant by "actual first page" right?

Reply:
Only one page needs to prerender. Otherwise we're wasting bandwidth/CPU. If the first page prerenders even though you have a `pageId` hash then your code needs to be fixed

Reply:
Ok that should be what I'm doing right now